vowpalwabbit/parser.cc

```diff
@@ -999,6 +999,17 @@ void free_parser(vw& all)
     output->currentname.delete_v();
   }
 
+  while (! all.p->example_pool.empty())
```

Comment:
Is it possible to get into a weird state where parser is deallocated, but there is still a thread trying to pass some of these through learn?

(Do we define that as improper consumption of the library?)

Reply:
It shouldn't happen given the way shutdown occurs.

Reply:
Long term it would be better if the pool took a deallocate function and used that when its destructor is called